src/coreclr/src/vm/threads.cpp

```diff
@@ -2627,6 +2627,13 @@ Thread::~Thread()
     }
 #endif // FEATURE_EVENT_TRACE
 
+#ifdef FEATURE_PERFTRACING
+    if (EventPipeThread::Get() != nullptr)
```

Comment:
I do not think `~Thread` is guaranteed to be called on its "owning" thread. It is often called on a different thread long after the physical thread that it was attached to died.